### PR TITLE
CB-18976 Changing network causes failure in PG upgrade

### PR DESCRIPTION
**Pull request by schfeca75**
CB-18976 Changing network causes failure in PG upgrade

In case of azure we will update the database subnets before executing the database upgrade in redbeams.

### CODE REVIEW

**Review by horadla23 — APPROVED** [review]

**Comment by gergopapi2** [comment]
The commit message describes the remedy introduced in this commit. Please add a short description of the problem, why the remedy is needed?

File: redbeams/src/main/java/com/sequenceiq/redbeams/service/stack/RedbeamsUpgradeService.java [hunk 1]
```diff
@@ -49,12 +66,27 @@ public void upgradeDatabaseServer(String crn, UpgradeDatabaseRequest upgradeData
...
         flowManager.notify(RedbeamsUpgradeEvent.REDBEAMS_START_UPGRADE_EVENT.selector(), redbeamsStartUpgradeRequest);
     }
 
+    private DBStack updateNetworkSubnets(DBStack dbStack) {
+        String cloudPlatform = dbStack.getCloudPlatform();
+        if (cloudParameterCache.isDbSubnetsUpdateEnabled(cloudPlatform)) {
+            DetailedEnvironmentResponse environment = environmentService.getByCrn(dbStack.getEnvironmentId());
+            Map<String, Object> envSubnets = networkBuilderService.getSubnetsFromEnvironment(environment, CloudPlatform.valueOf(cloudPlatform), dbStack);
+            Map<String, Object> networkAttributes = dbStack.getNetwork().getAttributes().getMap();
+            networkAttributes.putAll(envSubnets);
```
[the remark this answers is not in the document]
Comment (gergopapi2) [reply redbeams/src/main/java/com/sequenceiq/redbeams/service/stack/RedbeamsUpgradeService.java:82]: As discussed in person, this is not the case, so the code is ok.

File: redbeams/src/main/java/com/sequenceiq/redbeams/service/stack/RedbeamsUpgradeService.java [hunk 2]
```diff
@@ -49,12 +66,27 @@ public void upgradeDatabaseServer(String crn, UpgradeDatabaseRequest upgradeData
...
+            dbStack.getNetwork().setAttributes(new Json(networkAttributes));
+            return dbStackService.save(dbStack);
```
Comment (gergopapi2) [inline redbeams/src/main/java/com/sequenceiq/redbeams/service/stack/RedbeamsUpgradeService.java:84]: Please add a log line that network attributes were updated with env subnets, and the names could also be listed.

**Review by gergopapi2 — CHANGES_REQUESTED** [review]

**Review by gergopapi2 — APPROVED** [review]

**Comment by schfeca75** [comment]
> The commit message describes the remedy introduced in this commit. Please add a short description of the problem, why the remedy is needed?

Commit message has been updated

**Pull request merged**